tools/cmake/toolchains/GCC_ARM.cmake

```diff
@@ -7,6 +7,14 @@ set(CMAKE_CXX_COMPILER "arm-none-eabi-g++")
 set(GCC_ELF2BIN "arm-none-eabi-objcopy")
 set_property(GLOBAL PROPERTY ELF2BIN ${GCC_ELF2BIN})
 
+# Set definitions needed by Mbed OS
+function(mbed_set_toolchain_defintions target)
```

Comment:
I wonder, similar to above my question about setting gcc variable, isn't this all Mbed toolchain file for GCC ? If there is any generic info, it should be somewhere else?

For instance, setting TOOLCHAIN_ variables could be set somewhere else in the common file:

```
# pseudocode
if toolchain == this
set this symbols Mbed needs
elseif()
endif()
```

This way, this GCC_ARM.cmake would be Mbed specific toolchain file for Gcc Arm. Nothing generic in here. if 3rd party toolchain, do not use this file at all.

Create `generic_mbed_toolchain.cmake` or something ?

Reply:
I am not sure what you mean here either.
`GCC_ARM.cmake` is a Mbed specific toolchain file.

Reply:
I am confused. if 3rd party toolchain file is used, we should not include this gcc cmake.
The function `mbed_set_toolchain_defintions`  does not exist then does it? My point was to move this function elsewhere and keep gcc cmake toolchain specific file

Reply:
This should be somewhere else then not in our toolchain file:

```
# Set definitions needed by Mbed OS
function(mbed_set_toolchain_defintions target)
    target_compile_definitions(${target}
        INTERFACE
            TOOLCHAIN_GCC_ARM
            TOOLCHAIN_GCC
    )
endfunction()
```

Reply:
My proposal would be to have this in Mbed OS one, extending mbed-core target depending on a target - we can already check what toolchain is it with CMake variables? This is specific to Mbed OS sources so could be moved there

Reply:
I understand what you meant now and see the problem. Thanks for clarifying.

Reply:
Please review [this](https://github.com/ARMmbed/mbed-os/compare/222bfda751e386a4d94b5861d7e6c074e16e9a06..7608367a1a98ccf53706029664f817e88b7f6fb8).